src/composites.rs

```diff
@@ -239,6 +239,52 @@ where
     }
 }
 
+#[cfg(feature = "uds")]
+impl<'a, ReqBody, ResBody, Error, MakeError> Service<&'a tokio::net::UnixStream>
+    for CompositeMakeService<Option<SocketAddr>, ReqBody, ResBody, Error, MakeError>
+where
+    ReqBody: 'static,
+    ResBody: NotFound<ResBody> + 'static,
+    MakeError: Send + 'static,
+    Error: 'static,
+{
+    type Error = MakeError;
+    type Response = CompositeService<ReqBody, ResBody, Error>;
+    type Future = BoxFuture<'static, Result<Self::Response, Self::Error>>;
+
+    fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        for service in &mut self.0 {
+            match service.1.poll_ready(cx) {
+                Poll::Ready(Ok(_)) => {}
+                Poll::Ready(Err(e)) => {
+                    return Poll::Ready(Err(e));
+                }
+                Poll::Pending => {
+                    return Poll::Pending;
+                }
+            }
+        }
+        Poll::Ready(Ok(()))
+    }
+
+    fn call(&mut self, target: &'a tokio::net::UnixStream) -> Self::Future {
+        let mut services = Vec::with_capacity(self.0.len());
+        for (path, service) in &mut self.0 {
+            let path: &'static str = path;
+            services.push(
+                service
+                    .call(None)
```

Comment:
We _could_ pass `Some(target.peer_addr())` for consistency with the `TCP` equivalent - many middlewares just ignore the passed `target` so would still work.  Though any that actually care what the target is will not work with UDS sockets...

Reply:
Obviously the inner `Service` would need to be `Service<&'a tokio::net::uds::SocketAddr>`

Reply:
We could - in practice, existing middleware takes `Option<std::net::SocketAddr>` and the peer address is unlikely to be interesting anyway.